File: app2/src/main/java/com/ssz/framejava/app/AppStatus.java
package com.ssz.framejava.app;

import android.support.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * @author : zsp
 * time : 2019 11 2019/11/6 16:36
 * 程序状态
 */
@Retention(RetentionPolicy.SOURCE)
@StringDef({AppStatus.KILLED,AppStatus.ALIVE})
public @interface AppStatus {
    String KILLED = "killed";
    String ALIVE = "alive";
}

File: app/src/main/java/com/ssz/studydemo/app/AppStatus.kt
package com.ssz.studydemo.app

import android.support.annotation.StringDef

/**
 * @author : zsp
 * time : 2019 12 2019/12/11 13:20
 */
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@StringDef(AppStatus.KILLED, AppStatus.ALIVE)
annotation class AppStatus {
    companion object{
       const val KILLED = "killed"
       const val ALIVE = "alive"
    }
}